lnwallet/btcwallet/btcwallet.go

```diff
@@ -290,6 +290,39 @@ func (b *BtcWallet) IsOurAddress(a btcutil.Address) bool {
 	return result && (err == nil)
 }
 
+// ListWalletAddresses returns all addresses of addrType associated with the wallet.
+func (b *BtcWallet) ListWalletAddresses(addrType lnwallet.AddressType) ([]btcutil.Address, error) {
+	var (
+		keyScope        waddrmgr.KeyScope
+		walletAddresses []btcutil.Address
+	)
+
+	switch addrType {
+	case lnwallet.WitnessPubKey:
+		keyScope = waddrmgr.KeyScopeBIP0084
+	case lnwallet.NestedWitnessPubKey:
+		keyScope = waddrmgr.KeyScopeBIP0049Plus
+	default:
+		return nil, fmt.Errorf("unknown address type")
+	}
+
+	walletAccounts, err := b.wallet.Accounts(keyScope)
+	if err != nil {
+		return walletAddresses, err
```

Comment:
Could also just return `nil, err`, because the assumption is that the caller won't use the address set on error.